src/HttpReply.cc

```diff
@@ -456,6 +458,38 @@ HttpReply::parseFirstLine(const char *blk_start, const char *blk_end)
     return sline.parse(protoPrefix, blk_start, blk_end);
 }
 
+size_t
+HttpReply::parseTerminatedPrefix(const char * const terminatedBuf, const size_t bufSize)
+{
+    auto error = Http::scNone;
+    const bool eof = false; // TODO: Remove after removing atEnd from HttpHeader::parse()
+    if (parse(terminatedBuf, bufSize, eof, &error)) {
+        debugs(58, 7, "success after accumulating " << bufSize << " bytes and parsing " << hdr_sz);
+        Assure(pstate == Http::Message::psParsed);
+        Assure(hdr_sz > 0);
```

Comment:
If parse() returning true always implies hdr_sz>0 (as invariant) - probably we should check this invariant it inside parse() itself?

Reply:
> we should check this invariant it inside parse() itself

We do check it already -- there is an equivalent assert(hdr_len > 0) there, with some additional code that makes sure that hdr_sz is hdr_len when the method returns true. That code should be polished (and there is even an XXX related to that method in this PR), but that polishing should result in several out-of-scope Http::Message::parse() changes. Personally, I would rather not add Http::Message::parse() modifications to this PR, to avoid increasing its footprint and creating a false impression that we are changing how HTTP messages are parsed.

If others want this PR to adjust Http::Message::parse() despite my (weak/handwaving) arguments, I will add that assertion with the corresponding polishing touches.